src/pages/Register.tsx

```diff
@@ -0,0 +1,215 @@
+}).required();
+export const Register = () => {
+
+    const { register, handleSubmit, formState: { errors } } = useForm<RegisterData>({
+        resolver: yupResolver(schema),
+    });
+
+    const onSubmit = async (data: RegisterData) => {
+        console.log('Data being sent to backend:', data);
+        const jsonData = {
+            fname: data.firstName,
+            lname: data.lastName,
+            email: data.email,
+            password: data.password,
+        };
+
+        try {
+            const response = await axios.post('http://localhost:8080/signup', jsonData, {
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            });
+            console.log('Response:', response);  // Log the response to check
+            if (response.status === 200) {
+                toast.success('Registration successful');
+            } else {
+                toast.error('Registration failed');
+            }
+        } catch (error) {
+            console.error('Error during registration:', error); // Log error for debugging
+            toast.error('Registration failed. Check the server.');
+        }
+    };
+
+
+    return (
+        <Container
+            fluid
+            className={`d-flex justify-content-center align-items-center ${Object.keys(errors).length > 0 ? 'mt-5' : 'mt-4'
+                } mb-4`}
+        >
+            <Row className="w-100">
+                {/* Left Section */}
+                <Col lg={6} className="d-none d-lg-flex justify-content-center align-items-center">
+                    <img
+                        src="signup.jpeg"
+                        alt="Logo"
+                        className="img-fluid"
+                        style={{
+                            maxHeight: '80%',
+                            maxWidth: '90%',
+                            marginLeft: '10%',
+                        }}
+                    />
+                </Col>
+
+                {/* Right Section */}
+                <Col
+                    lg={6}
+                    xs={12}
+                    className="d-flex justify-content-center align-items-center mt-4 mb-4"
+                >
+                    <div
+                        className="p-4 shadow rounded position-relative card-with-corners rounded-4"
+                        style={{
+                            maxWidth: '100%',
+                            marginLeft: 'auto',
+                            marginRight: 'auto',
+                        }}
+                    >
+                        <Form onSubmit={handleSubmit(onSubmit)}>
+                            <h4 className="text-center mt-2">Sign Up</h4>
+                            {/* Social Icons */}
+                            <div className="d-flex justify-content-center gap-3 my-3">
+                                <Button variant="outline-primary" className="rounded-circle p-2">
+                                    <svg
```

Comment:
Move this svg to a seperate file and name icon-google.svg and place it under src/assets/icons